### DISCUSSION TITLE
Using Skeleton from Svelte

### DISCUSSION THREAD

**Discussion by sandeep-gh** [skeletonlabs/skeleton#2287] [General]
I have build a Python webdev framework with Svelte+Tailwind+Websocket+Starlette tech stack (https://github.com/ofjustpy/ofjustpy). I am looking into incorporating a Svelte UI library and Skeleton seems like the most obvious choice. I understand that the components are build using Tailwind and Svelte. But the examples I have seen so far seems that we use the components from TypeScript. Is it not possible to grab .Svelte file for a component and incorporate that into my browser runtime?

**Comment by endigo9740** [comment]
@sandeep-gh I'm not sure I quite understand the question here. Svelte components can be built using either Javascript or Typescript. We opt for Typescript for the obvious type safety that brings to the table. Please also note that we only provide support for SvelteKit, not Vite/Svelte or other manners of using Svelte. It's not that Skeleton can't support these (it can), it's just we're a small team and we focus on the largest majority of our audience - which opt for SvelteKit.

I would recommend the following:

- Make sure you understand how Svelte is being run in your environment (I'm assuming Vite is involved)
- Make sure you configure it to use the Typescript compiler, if possible, otherwise you'll have to adjust for JS yourself. This typically just means stripping out the types.
- If you need help with injecting components at run time, that may be a question to raise on the Svelte support forums. It's way out of scope of Skeleton.

Thanks!
